Doc/library/annotationlib.rst

```diff
@@ -510,6 +510,72 @@ annotations from the class and puts them in a separate attribute:
          return typ
 
 
+
+Creating a custom callable annotate function
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Custom :term:`annotate functions <annotate function>` may be literal functions like those
+automatically generated for functions, classes, and modules. Or, they may wish to utilise
+the encapsulation provided by classes, in which case any :term:`callable` can be used as
+an :term:`annotate function`.
+
+However, :term:`methods <method>`, class instances that implement
+:meth:`object.__call__`, and most other callables, do not provide the same attributes as
+true functions, which are needed for the :attr:`~Format.VALUE_WITH_FAKE_GLOBALS`
+machinery to work. :func:`call_annotate_function` and other :mod:`annotationlib`
+functions will attempt to infer those attributes where possible, but some of them must
+always be present for :attr:`~Format.VALUE_WITH_FAKE_GLOBALS` to work.
+
+Below is an example of a callable class that provides the necessary attributes to be
+used with all formats, and takes advantage of class encapsulation:
+
+.. code-block:: python
+
+  class Annotate:
+      called_formats = []
+
+      def __call__(self, format=None, *, _self=None):
```

Comment:
Is this the signature we want to force?

Reply:
Well, we're not really forcing it, but you're right that it is a somewhat awkward suggestion. I'll have a think about alternatives.

We could also move the logic in `__call__` to a separate method and then do the `self` and `_self` handling within `__call__` before passing to the method.